### PR TITLE
[exporter/sumologicexporter]: Added timeout validation 

### PR DESCRIPTION
**Pull request by chan-tim-sumo**
**Description:** <Describe what has changed.>
- Default timeout is 30 seconds with a max limit of 55. There're no validation checks for when [timeout is more than 55 seconds](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sumologicexporter/README.md?plain=1#L104). 

**Link to tracking Issue:** 
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33151 

**Testing:** 
- Unit test

**Documentation:** 
- chloggen

### CODE REVIEW

**Comment by chan-tim-sumo** [comment]
I think the easy cla check is still failing because I didn't set global user email in git until the most recent commit 🤔

**Comment by sumo-drosiek** [comment]
@chan-tim-sumo Please squash commits, maybe it will help with the CLA bot

**Review by sumo-drosiek — APPROVED** [review]
LGTM

**Review by swiatekm — APPROVED** [review]

**Comment by chan-tim-sumo** [comment]
oh i think i know why, even though after rebasing to one commit, the first commit's email is: 
```
Author: Tim Chan <timothy.chan@timothy.chan-mac>
```

after amending the author to sumo email:
```
Author: Tim Chan <timothy.chan@sumologic.com>
```

**Comment by chan-tim-sumo** [comment]
can anyone from @open-telemetry/collector-contrib-approvers take a look at this? 😄

**Comment by chan-tim-sumo** [comment]
hey @songy23, found you as one of the approvers, can you take a look? thanks!

File: exporter/sumologicexporter/config.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ func createDefaultClientConfig() confighttp.ClientConfig {
 
 func (cfg *Config) Validate() error {
 
+	if cfg.ClientConfig.Timeout < 0 || cfg.ClientConfig.Timeout > maxTimeout {
+		return fmt.Errorf("timeout must be between 0 and 55 seconds, got %v", cfg.ClientConfig.Timeout)
+	}
```
Comment (crobert-1) [inline exporter/sumologicexporter/config.go:78]: The underlying dependency that's being used is go's `http.Client` which is [created from this client config](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/73e67ba425481b94f51793df503b2e2b2e619514/exporter/sumologicexporter/exporter.go#L230), from what I can tell.

According to [documentation](https://pkg.go.dev/net/http#Client): 
```
	// A Timeout of zero means no timeout.
```
Should we explicitly not allow 0 to be a valid timeout then, if we want 55 seconds to be the max?

**Comment by chan-tim-sumo** [comment]
hey @dmitryax, i saw that you're one of the reviewers needed, can you take a look at this pr? thanks!

File: exporter/sumologicexporter/config.go (hunk 1 above)
Comment (chan-tim-sumo) [reply exporter/sumologicexporter/config.go:78]: ohh good catch, yeah that would be better. will make it between 1 to 55 seconds instead

**Review by crobert-1 — APPROVED** [review]

**Comment by crobert-1** [comment]
_If a code owner can check to ensure the new minimum timeout limit looks good, I can add `ready to merge`. Just want to make sure the change lines up with others' understanding as well._ 👍

**Review by rnishtala-sumo — APPROVED** [review]
LGTM!

**Review by sumo-drosiek — APPROVED** [review]

**Comment by chan-tim-sumo** [comment]
@dmitryax can you take a look at this? ready to merge :D

**Review by codeboten — APPROVED** [review]

**Pull request merged**